### ISSUE TITLE
Query :: consider removing unnecessary joins from queries generated during include collection

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFrameworkCore#6386]
Found during investigation of https://github.com/aspnet/EntityFramework/issues/6366

When we do complex includes like so:

``` cs
ctx.Posts
                    .Include(x => x.User).ThenInclude(x => x.Roles)
                    .Include(x => x.VoteDefinition).ThenInclude(x => x.PossibleAnswers)
```

we produce the following queries:

``` sql
SELECT [x].[Id], [x].[Name], [x].[UserId], [x].[VoteDefinitionId], [v].[Id], [v].[PostId], [u].[Id], [u].[Name]
FROM [Posts] AS [x]
LEFT JOIN [VoteDefinitions] AS [v] ON [v].[PostId] = [x].[Id]
INNER JOIN [Users] AS [u] ON [x].[UserId] = [u].[Id]
ORDER BY [v].[Id], [u].[Id]
```

``` sql
SELECT [r].[Id], [r].[Name], [r].[UserId]
FROM [Roles] AS [r]
INNER JOIN (
    SELECT DISTINCT [v].[Id], [u].[Id] AS [Id0]
    FROM [Posts] AS [x]
    LEFT JOIN [VoteDefinitions] AS [v] ON [v].[PostId] = [x].[Id]
    INNER JOIN [Users] AS [u] ON [x].[UserId] = [u].[Id]
) AS [u0] ON [r].[UserId] = [u0].[Id0]
ORDER BY [u0].[Id], [u0].[Id0]
```

``` cs
SELECT [v0].[Id], [v0].[VoteDefinitionId]
FROM [VoteDefinitionPossibleAnswers] AS [v0]
WHERE EXISTS (
    SELECT 1
    FROM [Posts] AS [x]
    LEFT JOIN [VoteDefinitions] AS [v] ON [v].[PostId] = [x].[Id]
    WHERE [v0].[VoteDefinitionId] = [v].[Id])
ORDER BY [v0].[VoteDefinitionId]
```

for the second query (the joined part) we dont need:

``` sql
LEFT JOIN [VoteDefinitions] AS [v] ON [v].[PostId] = [x].[Id]
```

since it's part of the second join path and is not relevant in the current context. We could/should strip all non-relevant JOINs from inner queries.

**Comment by AndriySvyryd** [comment]
@smitpatel Duplicate of #3419?

**Comment by smitpatel** [comment]
Closed fixed along with #12098
